### PR TITLE
chore: move packages to build-system-test/

### PR DESCRIPTION
**Pull request by 0618**
#### Description of changes

In https://github.com/aws-amplify/amplify-ui/pull/4057, I accidentally installed the package to monorepo...This PR is to fix that.

#### Description of how you validated changes

PoC PR https://github.com/aws-amplify/amplify-ui/pull/4077

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build-system-tests/package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (wlee221) [inline build-system-tests/package.json:1]: let's add `"private": "true"` 😈

File: build-system-tests/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  "name": "build-system-tests",
```
Comment (wlee221) [inline build-system-tests/package.json:2]:
```suggestion
  "name": "@aws-amplify/ui-build-system-tests",
```
not that this package is every installable or cross referenced, but just for consistency 🙏

File: build-system-tests/package.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
```
Comment (wlee221) [inline build-system-tests/package.json:10]:
```suggestion
```

**Review by wlee221 — COMMENTED** [review]
Just small nits, but looks good! 

Also, for my curiosity, is the reason we are using `npm` over `yarn` here is because `npm` is more commonly used?

**Comment by 0618** [comment]
> Just small nits, but looks good!
> 
> Also, for my curiosity, is the reason we are using `npm` over `yarn` here is because `npm` is more commonly used?

Right, plus when we test Build System, we only have 1 app tested with `yarn`. I'd be nice to keep it consistent inside `build-system-test/`

**Review by wlee221 — APPROVED** [review]

**Review by ioanabrooks — APPROVED** [review]

**Pull request merged**